addon/metrics-adapters/google-tag-manager.js

```diff
@@ -7,26 +7,42 @@ import canUseDOM from '../utils/can-use-dom';
 import objectTransforms from '../utils/object-transforms';
 import BaseAdapter from './base';
 
+const {
+  assert,
+  get,
+  set,
+  $,
+  getWithDefault,
+  isBlank,
+  String: { capitalize }
+} = Ember;
+const assign = Ember.assign || Ember.merge;
 const {
   compact
 } = objectTransforms;
 
 export default BaseAdapter.extend({
-  dataLayer: 'dataLayer',
+  dataLayerProp: 'dataLayer',
+  dataLayer: null,
 
   toStringExtension() {
     return 'GoogleTagManager';
   },
 
   init() {
     const config = get(this, 'config');
-    const { id, envParams } = config;
-    const dataLayer = getWithDefault(config, 'dataLayer', 'dataLayer');
+    const { id, envParams, dataLayer } = config;
+    if (!isBlank(dataLayer)) {
+      set(this, 'dataLayer', dataLayer);
+    }
+
+    const dataLayerProp = get(this, 'dataLayerProp');
+    const dataLayerValue = getWithDefault(config, dataLayerProp, []);
     const envParamsString = envParams ? `&${envParams}`: '';
 
     assert(`[ember-metrics] You must pass a valid \`id\` to the ${this.toString()} adapter`, id);
 
-    set(this, 'dataLayer', dataLayer);
+    window[dataLayerProp] = dataLayerValue;
```

Comment:
@bugduino I'm new to this but I've just noticed that it seems like you should not use the **window** if you are not sure you `canUseDom`. I would suggest to move it down a bit (put it below the `canUseDom`) to prevent unseen problems.